beginner_source/dist_overview.rst

```diff
@@ -15,107 +15,62 @@ to the technology that can best serve your use case.
-4. Use multi-machine `DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
+3. Use multi-machine `DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
    and the `launching script <https://github.com/pytorch/examples/blob/master/distributed/ddp/README.md>`__,
    if the application needs to scale across machine boundaries.
-5. Use multi-GPU `FullyShardedDataParallel <https://pytorch.org/docs/stable/fsdp.html>`__
+4. Use multi-GPU `FullyShardedDataParallel <https://pytorch.org/docs/stable/fsdp.html>`__
    training on a single-machine or multi-machine when the data and model cannot
    fit on one GPU.
-6. Use `torch.distributed.elastic <https://pytorch.org/docs/stable/distributed.elastic.html>`__
-   to launch distributed training if errors (e.g., out-of-memory) are expected or if
-   resources can join and leave dynamically during training.
 
 
 .. note:: Data-parallel training also works with `Automatic Mixed Precision (AMP) <https://pytorch.org/docs/stable/notes/amp_examples.html#working-with-multiple-gpus>`__.
 
 
-``torch.nn.DataParallel``
-~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The `DataParallel <https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html>`__
-package enables single-machine multi-GPU parallelism with the lowest coding
-hurdle. It only requires a one-line change to the application code. The tutorial
-`Optional: Data Parallelism <../beginner/blitz/data_parallel_tutorial.html>`__
-shows an example. Although ``DataParallel`` is very easy to
-use, it usually does not offer the best performance because it replicates the
-model in every forward pass, and its single-process multi-thread parallelism
-naturally suffers from
-`GIL <https://wiki.python.org/moin/GlobalInterpreterLock>`__ contention. To get
-better performance, consider using
-`DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__.
-
-
-``torch.nn.parallel.DistributedDataParallel``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Compared to `DataParallel <https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html>`__,
-`DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
-requires one more step to set up, i.e., calling
-`init_process_group <https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group>`__.
-DDP uses multi-process parallelism, and hence there is no GIL contention across
-model replicas. Moreover, the model is broadcast at DDP construction time instead
-of in every forward pass, which also helps to speed up training. DDP is shipped
-with several performance optimization technologies. For a more in-depth
-explanation, refer to this
-`paper <http://www.vldb.org/pvldb/vol13/p3005-li.pdf>`__ (VLDB'20).
-
-
 DDP materials are listed below:
```

Comment:
can we get rid of the `Single-Machine Model Parallel Best Practices` in this section (inside the second bullet point)?